orc8r/cloud/go/obsidian/swagger/handlers/handlers.go

```diff
@@ -33,3 +66,115 @@
 		return c.String(http.StatusOK, combined)
 	}
 }
+
+// GetSpecHandler returns a routing handler which creates and
+// serves the raw YAML spec of a singular service.
+func GetSpecHandler(yamlCommon string) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		requestedService := c.Param("service")
+
+		service, ok, err := getServiceName(requestedService)
+		if err != nil {
+			return obsidian.HttpError(err, http.StatusInternalServerError)
+		}
+		if !ok {
+			return obsidian.HttpError(errors.New("service not found"), http.StatusNotFound)
+		}
+
+		combined, err := swagger.GetCombinedSpecFromService(yamlCommon, service)
+		if err != nil {
+			return obsidian.HttpError(err, http.StatusInternalServerError)
+		}
+
+		return c.String(http.StatusOK, combined)
+	}
+}
+
+// SpecInfo defines the URL to be injected into
+// the Swagger UI template.
+type SpecInfo struct {
+	URL string
```

Comment:
"url" (unless making private doesn't play nicely with the templating pkg)

Reply:
Yeah both the field and the struct have to be public in order for it to properly template. The reason why I went with "URL" instead of "Url" is because I've never seen the latter being used.

Reply:
Gotcha good to know. So yeah then URL is correct